test/functional/testlib.bash

```diff
@@ -874,25 +874,32 @@ create_version() {
 # validate the swupd client
 # Parameters:
 # - -e: if this option is set the test environment is created empty (withouth bundle os-core)
+# - -r: if this option is set the test environment is created with a more complete version of
+#       the os-core bundle that includes release files, it is useful for some tests like update tests
 # - ENVIRONMENT_NAME: the name of the test environment, this should be typically the test name
 # - VERSION: the version to use for the test environment, if not specified the default is 10
 # - FORMAT: the format number to use initially in the environment
 create_test_environment() { 
 
 	local empty=false
+	local release_files=false
 	[ "$1" = "-e" ] && { empty=true ; shift ; }
+	[ "$1" = "-r" ] && { release_files=true ; shift ; }
```

Comment:
what if `-r` comes before `-e`? Will `-e` be ignored?

Reply:
Yes. It is a limitation of using parameters this way, but the alternative is using a full arguments parser (getups) like create_bundle function does, which in my opinion is just too heavy for this. I'm trying to stick to simple arguments parser unless the function is too complex (lots of parameters) like create_bundle.
Also, in this case you cannot have an empty environment and one with a complex os-core, they are mutually exclusive.

Reply:
I guess I could use an if ... elif clause to improve those function arguments...

Reply:
Or you could just document the limitation in the function comment and readme, I'd be fine with that.

Reply:
I'll fix that

Reply:
never mind, now that I think of it, using an if elif clause would be the same result, so the only way I can think of for not overriding those parameters would be using getopts, and I don't think is a good idea since these functions get called with every single test, it would slow down execution.

Reply:
The usage info also shows that you could use one parameter or the other:
 create_test_environment [-e|-r]

Reply:
@matthewrsj I updated the usage info in both functions to show the limitation when using the parameters. I didn't added it to the README because I found it to be very obvious there (not being able to create an empty environment and with a os-core bundle at the same time). If you still think is necessary to add something to the readme, just let me know what kind of message you would like to see there and I'll add it.